### PR TITLE
Fix regression: Omit version property dependency

### LINKED ISSUE #1869: Regression: Exception while restoring on mono
```
Restoring packages for /var/www/Code/dylannet/vnext/dylan.NET.Tasks/project.json
----------
Microsoft.Framework.Runtime.FileFormatException: Object reference not set to an instance of an object ---> Microsoft.Framework.Runtime.FileFormatException: Object reference not set to an instance of an object ---> System.NullReferenceException: Object reference not set to an instance of an object
  at Microsoft.Framework.Runtime.Project.PopulateDependencies (System.String projectPath, IList`1 results, Microsoft.Framework.Runtime.Json.JsonObject settings, System.String propertyName, Boolean isGacOrFrameworkReference) <0xb3780948 + 0x00263> in <filename unknown>:0 
  at Microsoft.Framework.Runtime.Project.BuildTargetFrameworkNode (System.String frameworkKey, Microsoft.Framework.Runtime.Json.JsonObject frameworkValue) <0xb377f000 + 0x00277> in <filename unknown>:0 
  at Microsoft.Framework.Runtime.Project.BuildTargetFrameworksAndConfigurations (Microsoft.Framework.Runtime.Json.JsonObject projectJsonObject) <0xb379cd08 + 0x004d3> in <filename unknown>:0 
  --- End of inner exception stack trace ---
  at Microsoft.Framework.Runtime.Project.BuildTargetFrameworksAndConfigurations (Microsoft.Framework.Runtime.Json.JsonObject projectJsonObject) <0xb379cd08 + 0x00523> in <filename unknown>:0 
  at Microsoft.Framework.Runtime.Project.GetProjectFromStream (System.IO.Stream stream, System.String projectName, System.String projectPath, ICollection`1 diagnostics) <0xb37945a8 + 0x01353> in <filename unknown>:0 
  at Microsoft.Framework.Runtime.Project.TryGetProject (System.String path, Microsoft.Framework.Runtime.Project& project, ICollection`1 diagnostics) <0xb3794230 + 0x000cf> in <filename unknown>:0 
  --- End of inner exception stack trace ---
  at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw () <0xb4206eb0 + 0x00035> in <filename unknown>:0 
  at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess (System.Threading.Tasks.Task task) <0xb4205f10 + 0x000b7> in <filename unknown>:0 
  at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification (System.Threading.Tasks.Task task) <0xb4205e80 + 0x00084> in <filename unknown>:0 
  at System.Runtime.CompilerServices.TaskAwaiter.ValidateEnd (System.Threading.Tasks.Task task) <0xb4205e30 + 0x0003f> in <filename unknown>:0 
  at System.Runtime.CompilerServices.TaskAwaiter`1[TResult].GetResult () <0xb44f96b0 + 0x0001f> 24189 in <filename unknown>:0 
  at Microsoft.Framework.PackageManager.RestoreCommand+<>c__DisplayClass72_0+<<ExecuteCommand>b__0>d.MoveNext () <0xb378fc60 + 0x00157> in <filename unknown>:0 
--- End of stack trace from previous location where exception was thrown ---
  at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw () <0xb4206eb0 + 0x00035> in <filename unknown>:0 
  at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess (System.Threading.Tasks.Task task) <0xb4205f10 + 0x000b7> in <filename unknown>:0 
  at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification (System.Threading.Tasks.Task task) <0xb4205e80 + 0x00084> in <filename unknown>:0 
  at System.Runtime.CompilerServices.TaskAwaiter.ValidateEnd (System.Threading.Tasks.Task task) <0xb4205e30 + 0x0003f> in <filename unknown>:0 
  at System.Runtime.CompilerServices.TaskAwaiter.GetResult () <0xb4205e00 + 0x0001f> in <filename unknown>:0 
  at Microsoft.Framework.PackageManager.RestoreCommand+<ExecuteCommand>d__72.MoveNext () <0xb3930108 + 0x00787> in <filename unknown>:0 
----------
Restore failed
Object reference not set to an instance of an object
```

### PR DESCRIPTION
**Pull request by troydai**
Fix #1869 

/cc @davidfowl @borgdylan

### CODE REVIEW

File: src/Microsoft.Framework.Runtime/Project.cs [hunk 1]
```diff
@@ -379,7 +379,7 @@ private static SemanticVersion SpecifySnapshot(string version, string snapshotVa
                     }
 
                     SemanticVersionRange dependencyVersionRange = null;
-                    if (!string.IsNullOrEmpty(dependencyVersionAsString.Value))
+                    if (dependencyVersionAsString != null && !string.IsNullOrEmpty(dependencyVersionAsString.Value))
```
Comment (davidfowl) [inline src/Microsoft.Framework.Runtime/Project.cs]: ```string.IsNullOrEmpty(dependencyVersionAsString.Value)```

Comment (ChengTian) [inline src/Microsoft.Framework.Runtime/Project.cs]: +1
I guess David means `string.IsNullOrEmpty(dependencyVersionAsString?.Value)``

**Comment by ChengTian** [comment]
I build this branch locally and verified that it has the same behavior as what we had before internalization of JSON parser:

`dnu restore` ignores the dependency that doesn't have `version`, like:
```
"MyDependency" : {"type": "build"}

```
:shipit: after minor changes according to comments above.

File: src/Microsoft.Framework.Runtime/Project.cs (hunk 1 above)
Comment (troydai) [inline src/Microsoft.Framework.Runtime/Project.cs]: Yeah, this is nicer.

**Comment by davidfowl** [comment]
:shipit:

**Comment by troydai** [comment]
I'll merge soon.

**Pull request merged**